assembly/buffer/index.ts

```diff
@@ -46,6 +46,50 @@ export class Buffer extends Uint8Array {
 
   readInt8(offset: i32 = 0): i8 {
     if(<u32>offset >= this.dataLength) throw new RangeError(E_INDEXOUTOFRANGE);
-    return load<i8>(this.dataStart + usize(offset));
+    return load<i8>(this.dataStart + <usize>offset);
+  }
+
+  readInt16LE(offset: i32 = 0): i16 {
+    if(<u32>offset >= this.dataLength - 1) throw new RangeError(E_INDEXOUTOFRANGE);
```

Comment:
If `this.dataLength = 0`, this wraps around to `offset >= u32.MAX_VALUE`, which is always false. From std DataView:

```ts
    if (
      i32(offset < 0) |
      i32(byteOffset + 2 > this.dataLength)
    ) throw new RangeError(E_INDEXOUTOFRANGE);
```

Reply:
When thinking about it, DataView isn't correct for this as well. If `offset = i32.MAX_VALUE` for example, the second part will wrap around to `-2147483647 > this.dataLength` which is always `false` as well. Proper version is a bit uglier:

```ts
    if (
      i32(offset < 0) |
      i32(<u32>offset + 2 > <u32>this.dataLength)
    ) throw new RangeError(E_INDEXOUTOFRANGE);
```

Reply:
I think I asked about this a few times, but was told to follow the std lib examples. Looks like we should be doing explicit `offset < 0` checks as @RedDwarfian proposed initially :D.

Think you can make that change on every function so far?

Reply:
I'll get that done when I get home from work tonight.